File: utilities/src/main/java/com/toocol/termio/utilities/escape/EscapeCodeSequenceSupporter.java
package com.toocol.termio.utilities.escape;

import com.toocol.termio.utilities.escape.actions.AnsiEscapeAction;

import java.util.List;

/**
 * @author ：JoeZane (joezane.cn@gmail.com)
 * @date: 2022/8/16 23:54
 * @version: 0.0.1
 */
public interface EscapeCodeSequenceSupporter<T> {

    List<AnsiEscapeAction<T>> registerActions();

    void printOut(String text);

}

File: utilities/src/main/java/com/toocol/termio/utilities/escape/EscapeCodeSequenceSupporter.kt
package com.toocol.termio.utilities.escape

/**
 * @author ：JoeZane (joezane.cn@gmail.com)
 * @date: 2022/8/16 23:54
 * @version: 0.0.1
 */
interface EscapeCodeSequenceSupporter<T> {
    fun getActionMap(): Map<Class<out IEscapeMode>, AnsiEscapeAction<T>>?

    fun printOut(text: String)
}